x/emissions/keeper/inference_synthesis/synth_palette_weight.go

```diff
@@ -201,14 +201,24 @@ func (p *SynthPalette) CalcWeightedInference(weights RegretInformedWeights) (Inf
 	return ret, nil
 }
 
-func (p *SynthPalette) GetInfererRegretsSlice() []alloraMath.Dec {
+func (p *SynthPalette) GetInfererRegretsSliceAndSetFlags() []alloraMath.Dec {
 	var regrets []alloraMath.Dec
 	if len(p.InfererRegrets) == 0 {
 		return regrets
 	}
 	regrets = make([]alloraMath.Dec, 0, len(p.InfererRegrets))
-	for _, worker := range p.InfererRegrets {
-		regrets = append(regrets, worker.regret)
+	p.InferersNewStatus = InferersAllNew
+	for address, regretInfo := range p.InfererRegrets {
+		if !regretInfo.noPriorRegret {
+			if p.InferersNewStatus == InferersAllNew {
+				p.InferersNewStatus = InferersAllNewExceptOne
+				p.SingleNotNewInferer = address
+			} else {
+				p.InferersNewStatus = InferersNotNew
+				p.SingleNotNewInferer = ""
+			}
+		}
+		regrets = append(regrets, regretInfo.regret)
```

Comment:
could maybe refactor this into a helper function somewhere since I think it is used in multiple places in the codebase, but I'm flexible